ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -100,7 +100,7 @@ ghcLibParserHsSrcDirs forParserDepends ghcFlavor lib =
         -- 'GhcVersion.hs' IIRC).
         [ stage0GhcBoot | ghcFlavor `elem` [ GhcMaster, Ghc8101, Ghc8102, Ghc901 ] ] ++
         map takeDirectory cabalFileLibraries ++
-        askFiles lib "hs-source-dirs:"
+        filter (not . isSuffixOf ".") (askFiles lib "hs-source-dirs:")
```

Comment:
I'm not convinced this is the right approach. For instance, if we only have `foo/.` in the list but not `foo`, then there will be no `foo` left afterward. I think doing what was suggested in the issue and normalizing file paths seems more appropriate. [`System.FilePath.normalise`](https://hackage.haskell.org/package/filepath-1.4.2.1/docs/System-FilePath-Posix.html#v:normalise) could be very helpful for this.

Reply:
I looked into this very deeply. I won't bother you with the details of why but the fix is correct (but am happy to do so if you want an explanation). That said I have no problems with throwing a `normalise` in the mix so stand by for that.

Reply:
I agree with you that using `normalise` is more robust and therefore better for future proofing.